wen-restart/src/last_voted_fork_slots_aggregate.rs

```diff
@@ -0,0 +1,245 @@
+            active_percent,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        crate::{
+            last_voted_fork_slots_aggregate::LastVotedForkSlotsAggregate,
+            solana::wen_restart_proto::{
+                LastVotedForkSlotsAggregateRecord, LastVotedForkSlotsRecord,
+            },
+        },
+        solana_gossip::restart_crds_values::RestartLastVotedForkSlots,
+        solana_runtime::{
+            bank::Bank,
+            genesis_utils::{
+                create_genesis_config_with_vote_accounts, GenesisConfigInfo, ValidatorVoteKeypairs,
+            },
+        },
+        solana_sdk::{hash::Hash, signature::Signer, timing::timestamp},
+        std::collections::HashMap,
+    };
+
+    #[test]
+    fn test_aggregate() {
+        solana_logger::setup();
+        let validator_voting_keypairs: Vec<_> =
+            (0..10).map(|_| ValidatorVoteKeypairs::new_rand()).collect();
+        let GenesisConfigInfo { genesis_config, .. } = create_genesis_config_with_vote_accounts(
+            10_000,
+            &validator_voting_keypairs,
+            vec![100; validator_voting_keypairs.len()],
+        );
+        let (_, bank_forks) = Bank::new_with_bank_forks_for_tests(&genesis_config);
+        let root_bank = bank_forks.read().unwrap().root_bank();
+        let root_slot = root_bank.slot();
+        let shred_version = 52;
+        let mut slots_vec = Vec::new();
+        let last_voted_fork = vec![root_slot + 1, root_slot + 2, root_slot + 3];
+        let mut slots_aggregate = LastVotedForkSlotsAggregate::new(
+            root_slot,
+            0.42,
+            root_bank.epoch_stakes(root_bank.epoch()).unwrap(),
+            &last_voted_fork,
+            &validator_voting_keypairs[9].node_keypair.pubkey(),
+        );
+        let mut expected_aggregate_record = HashMap::new();
+        for validator_voting_keypair in validator_voting_keypairs.iter().take(3) {
+            let pubkey = validator_voting_keypair.node_keypair.pubkey();
+            slots_vec.push(
+                RestartLastVotedForkSlots::new(
+                    pubkey,
+                    timestamp(),
+                    &last_voted_fork,
+                    Hash::default(),
+                    shred_version,
+                )
+                .unwrap(),
+            );
+            expected_aggregate_record.insert(
+                pubkey.to_string(),
+                LastVotedForkSlotsRecord {
+                    last_vote_fork_slots: last_voted_fork.clone(),
+                    last_vote_bankhash: Hash::default().to_string(),
+                    shred_version: shred_version as u32,
+                },
+            );
+        }
+        let mut aggregate_record = LastVotedForkSlotsAggregateRecord {
+            received: HashMap::new(),
+        };
+        let result = slots_aggregate.aggregate(slots_vec, &mut aggregate_record);
+        assert_eq!(result.active_percent, 40.0);
```

Comment:
Quantifying and making these calculations explicit would be more readable here.

For instance:

1. setting a variable `const TOTAL_VALIDATOR_COUNT = 10` equal to `validator_voting_keypairs.len()` and then 
2.  Set a variable `initial_num_active_validators = 3`, and using that above in `for validator_voting_keypair in validator_voting_keypairs.iter().take(num_active_validators)`. 
3. Then calculating this percentage based on `num_active_validators` and `TOTAL_VALIDATOR_COUNT`

Reply:
Done.